### PR TITLE
Create doc and scripts for qpl_deflate benchmark

### PR DESCRIPTION
**Pull request by jinjunzh**
### Changelog category (leave one):
- Documentation (changelog entry is not required)

### Documentation entry for user-facing changes
- Create documents and scripts for new "DeflateQpl" compression codec, include how to build & setup IAA  & run benchmark.

### CODE REVIEW

File: docs/en/sql-reference/statements/create/table.md [hunk 1]
```diff
@@ -379,8 +379,8 @@ High compression levels are useful for asymmetric scenarios, like compress once,
 `DEFLATE_QPL` — [Deflate compression algorithm](https://github.com/intel/qpl) implemented by Intel® Query Processing Library. Some limitations apply:
 
 -   DEFLATE_QPL is experimental and can only be used after setting configuration parameter `allow_experimental_codecs=1`.
--   DEFLATE_QPL only works if ClickHouse was compiled with support for AVX2 or AVX512 instructions
--   DEFLATE_QPL works best if the system has a Intel® IAA (In-Memory Analytics Accelerator) offloading device
+-   DEFLATE_QPL only works if ClickHouse was compiled with support for AVX2 or AVX512 instructions. Refer to [Build Tips](https://github.com/ClickHouse/ClickHouse/tree/master/contrib/qpl-cmake/doc/build.md) for more details.
```
Comment (rschu1ze) [inline docs/en/sql-reference/statements/create/table.md:382]: The linked documentation will currently not be rendered with ClickHouse's nice standard documentation template (e.g. https://clickhouse.com/docs/en/sql-reference/statements/create/table).

What about including documents build.md and benchmark.md in ClickHouse's contribution guide? E.g. we could merge both files and include them on the same level as "Third-Party Libraries" or "Integrating Rust libraries" under the title "Building and Benchmarking Intel QPL".

File: contrib/qpl-cmake/doc/build.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+This doc gives specfic build tips for QPL_Deflate codec based on Clickhouse generic [build instructions](https://github.com/ClickHouse/ClickHouse/blob/master/docs/en/development/build.md)
```
Comment (rschu1ze) [inline contrib/qpl-cmake/doc/build.md:1]: If this file is moved into the regular ClickHouse documentation (please see my below comment), we should replace the link from an absolute to a relative link. As an example for the necessary syntax, please see file "docs/en/sql-reference/functions/type-conversion-functions.md".

File: contrib/qpl-cmake/doc/build.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+Several tips to build Clickhouse with QPL_deflate enabled:
```
Comment (rschu1ze) [inline contrib/qpl-cmake/doc/build.md:3]: Hm, I guess there should be a more clear distinction between requirements of the QPL library and requirements of QPL codec support in ClickHouse. If I see correctly, the first three points affect only the QPL library. Maybe it makes sense to omit them and instead link to https://intel.github.io/qpl/documentation/get_started_docs/installation.html?

File: contrib/qpl-cmake/doc/build.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+- The QPL build only support x86_64 platform with avx2/avx512 support.
+- The QPL build requires nasm 2.15.0 or higher (e.g., can be obtained from https://www.nasm.us)
+- The QPL requires C++ compiler with C++17 standard support.
+- Pass the following flag to CMake. Which one to choose depends on your target hardware supported with AVX2 or AVX512.
```
Comment (rschu1ze) [inline contrib/qpl-cmake/doc/build.md:7]: "Pass the following flag to CMake when building ClickHouse, depending on the capabilities of your target machine."

File: contrib/qpl-cmake/doc/benchmark.md [hunk 5]
```diff
@@ -0,0 +1,270 @@
...
+
+"client_scripts" contains python scripts for running typical benchmark, for example:
+- "client_stressing_test.py": The python script for query stress test with [1~4] server instances.      
+- "queries_ssb.sql": The file lists all queries for [Star Schema Benchmark](https://clickhouse.com/docs/en/getting-started/example-datasets/star-schema/)
+- "allin1_ssb.sh": This shell script executes benchmark workflow all in one automatically.
+"database_files" means it will store database files according to lz4/deflate/zstd codecs.
+
+"rawdata_dir" means it will store raw data generated by dbgen.
+
+## Run benchmark automatically for Start Schema:
```
Comment (rschu1ze) [inline contrib/qpl-cmake/doc/benchmark.md:12]: Start --> Star

File: contrib/qpl-cmake/benchmark_sample/client_scripts/allin1_ssb.sh [hunk 6]
```diff
@@ -0,0 +1,530 @@
+#!/bin/bash
```
Comment (rschu1ze) [inline contrib/qpl-cmake/benchmark_sample/client_scripts/allin1_ssb.sh:1]: Wow, nice benchmark script.

**Review by rschu1ze — APPROVED** [review]
Very nice, thanks. I have only a few comments about where to store the docs. Besides that, I am happy to merge.

**Comment by rschu1ze** [comment]
The [docs check](https://s3.amazonaws.com/clickhouse-test-reports/47745/038b56ad6a752bf71ce0254877daf26f03b6d47f/style_check/docs_spelling_output.txt) has a (non-sensical) complaint:

```
====== /ClickHouse/docs/en/development/building_and_benchmarking_deflate_qpl.md ======
lineorder
====== Errors found ======
To exclude some words add them to the dictionary file "/ClickHouse/utils/check-style/aspell-ignore/en/aspell-dict.txt"
You can also run ./check-doc-aspell -i to see the errors interactively and fix them or add to the dictionary file
```

Could you please follow this recommendation? (GitHub won't let me do this little change by myself) Thanks!

**Comment by jinjunzh** [comment]
> The [docs check](https://s3.amazonaws.com/clickhouse-test-reports/47745/038b56ad6a752bf71ce0254877daf26f03b6d47f/style_check/docs_spelling_output.txt) has a (non-sensical) complaint:
> 
> ```
> ====== /ClickHouse/docs/en/development/building_and_benchmarking_deflate_qpl.md ======
> lineorder
> ====== Errors found ======
> To exclude some words add them to the dictionary file "/ClickHouse/utils/check-style/aspell-ignore/en/aspell-dict.txt"
> You can also run ./check-doc-aspell -i to see the errors interactively and fix them or add to the dictionary file
> ```
> 
> Could you please follow this recommendation? (GitHub won't let me do this little change by myself) Thanks!

Done

**Pull request merged**